d2ir/compile.go

```diff
@@ -396,20 +410,87 @@ func (c *compiler) compileKey(refctx *RefContext) {
 }
 
 func (c *compiler) compileField(dst *Map, kp *d2ast.KeyPath, refctx *RefContext) {
-	if refctx.Key != nil && len(refctx.Key.Edges) == 0 && refctx.Key.Value.Null != nil {
-		// For vars, if we delete the field, it may just resolve to an outer scope var of the same name
-		// Instead we keep it around, so that resolveSubstitutions can find it
-		if !IsVar(dst) {
-			dst.DeleteField(kp.IDA()...)
-			return
-		}
+	if refctx.Key.Ampersand {
+		return
 	}
-	f, err := dst.EnsureField(kp, refctx)
+
+	fa, err := dst.EnsureField(kp, refctx, true)
 	if err != nil {
 		c.err.Errors = append(c.err.Errors, err.(d2ast.Error))
 		return
 	}
 
+	for _, f := range fa {
+		c._compileField(f, refctx)
+	}
+}
+
+func (c *compiler) ampersandFilter(refctx *RefContext) bool {
+	if !refctx.Key.Ampersand {
+		return true
+	}
+	if len(refctx.Key.Edges) > 0 {
```

Comment:
is ampersand not functional for edges?

```d2
fast: {
  a
  far
}

task: {
  a
}

rats

* -> rats: {
  &label: task
}
```

![Screen Shot 2023-07-30 at 11 42 05 AM](https://github.com/terrastruct/d2/assets/3120367/207e322e-45aa-4b68-b4ae-4e0c397c3bb5)

Reply:
Ah I forgot will fix.

Reply:
Oh no so it does work on edge maps. But you can't use ampersands to filter maps with edges. E.g. you can't say apply this to a map with an edge from y -> z.

Reply:
Nvm there was a bug. fixed. See test.

Reply:
i don't understand. the test doesn't test this case. after latest pull, i now get `glob filters cannot be used outside globs` with the same code.

Reply:
I'm not sure what you're trying to do in your test there.